addons/lunch/models/lunch_supplier.py

```diff
@@ -0,0 +1,172 @@
+        ('mail', 'Email'),
+    ], 'Send Order By', default='phone')
+    automatic_email_send = fields.Boolean('Automatic Email Sending')
+    automatic_email_time = fields.Float('Hour')
+
+    recurrency = fields.Selection([('once', 'Specific Day'), ('reccurent', 'Reccurent')], 'Recurrency', default='once')
+    recurrency_from = fields.Float('From')
+    recurrency_to = fields.Float('To')
+    recurrency_date = fields.Date('Day', default=fields.Date.today())
+    recurrency_date_from = fields.Datetime('from', compute='_compute_recurrency_date_from', store=True)
+    recurrency_date_to = fields.Datetime('to', compute='_compute_recurrency_date_to', store=True)
+    recurrency_monday = fields.Boolean('Monday')
+    recurrency_tuesday = fields.Boolean('Tuesday')
+    recurrency_wednesday = fields.Boolean('Wednesday')
+    recurrency_thursday = fields.Boolean('Thursday')
+    recurrency_friday = fields.Boolean('Friday')
+    recurrency_saturday = fields.Boolean('Saturday')
+    recurrency_sunday = fields.Boolean('Sunday')
+
+    available_today = fields.Boolean('', compute='_compute_available_today', search='_search_available_today')
+
+    tz = fields.Selection(_tz_get, string='Timezone', required=True, default='UTC')
+
+    _sql_constraints = [
+        ('automatic time', 'CHECK(automatic_email_time >= 0 AND automatic_email_time <= 24)', 'Automatic Email Sending Time should be between 0 and 24'),
+        ('recurrency from', 'CHECK(recurrency_from >= 0 AND recurrency_from <= 24)', 'Recurrency From should be between 0 and 24'),
+        ('recurrency to', 'CHECK(recurrency_to >= 0 AND recurrency_to <= 24)', 'Recurrency To should be between 0 and 24')
+    ]
+
+    @api.model
+    def _auto_email_send(self):
+        records = self.search([('automatic_email_send', '=', True)])
+
+        for supplier in records:
+            send_at = datetime.combine(fields.Date.today(), float_to_time(supplier.automatic_email_time, supplier.tz)).astimezone(pytz.UTC).replace(tzinfo=None)
+            if supplier.available_today and fields.Datetime.now() > send_at:
+                orders = self.env['lunch.order'].search([('mail_sent', '=', False), ('supplier_ids', 'in', [supplier.id]), ('state', '=', 'ordered')])
+
+                if orders:
+                    order = {
+                        'company_name': orders[0].company_id.name,
+                        'currency_id': orders[0].currency_id.id,
+                        'supplier_id': supplier.partner_id.id,
+                        'supplier_name': supplier.name,
+                        'supplier_email': supplier.email_formatted
+                    }
+
+                    lines = orders.mapped('order_line_ids').filtered(lambda line: line.supplier == supplier)
+                    lines = [{'product': line.product_id.name, 'note': line.note, 'quantity': line.qty, 'price': line.price} for line in lines]
+
+                    order['amount_total'] = sum(l['price'] for l in lines)
```

Comment:
Is there multi currency in lunch ? if not, then it's ok.
But you can use `sum(lines.mapped('price'))`